src/main/java/org/wso2/carbon/esb/connector/MQPublishTopic.java

```
@@ -0,0 +1,134 @@
+            }
+
+            connectionBuilder.closeConnection();
+
+        } catch (Exception e) {
+            log.error("Problem in publishing to message" + e);
+            throw new ConnectException(e);
+        }
+    }
+
+    /**
+     * Initialize topic
+     */
+    MQTopic setTopic(MQConnectionBuilder connectionBuilder, MQConfiguration config) {
+        MQQueueManager manager = connectionBuilder.getManager();
+        MQTopic publisher;
+        try {
+            publisher = manager.accessTopic(config.getTopicString(), config.getTopicName(),
+                    CMQC.MQTOPIC_OPEN_AS_PUBLICATION, CMQC.MQOO_OUTPUT);
+        } catch (MQException e) {
+            log.error("Error creating topic" + e);
+            return null;
+        }
+        return publisher;
+    }
+
+    /**
+     * Create MQMessage
+     */
+    MQMessage buildMQMessage(MQConfiguration config, String queueMessage) {
+
+        MQMessage mqMessage = new MQMessage();
+        mqMessage.messageId = config.getMessageID().getBytes();
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
```

Returning null and handle error as generic, we will be loosing information why this got failed. It s better if we could throw this exception to higher layer and handle over there.

---

log.info("Error creating mq message" , e);